### PR TITLE
Commits from UNH plus fixes

### PR DESCRIPTION
**Pull request by strimper13**
"youre not getting a quote from me" Matt

### CODE REVIEW

File: src/main/java/frc/team166/robot/Robot.java [hunk 1]
```diff
@@ -111,9 +117,9 @@ public void teleopInit() {
         // teleop starts running. If you want the autonomous to
         // continue until interrupted by another command, remove
         // this line or comment it out.
-        if (m_autonomousCommand != null) {
-            m_autonomousCommand.cancel();
-        }
+        //     if (m_autonomousCommand != null) {
```
Comment (bot190) [inline src/main/java/frc/team166/robot/Robot.java]: We should uncomment this.

File: src/main/java/frc/team166/robot/Robot.java [hunk 2]
```diff
@@ -93,8 +99,8 @@ public void autonomousInit() {
 
         // schedule the autonomous command (example)
         if (m_autonomousCommand != null) {
-            m_autonomousCommand.start();
-        }
+             m_autonomousCommand.start();
```
Comment (bot190) [inline src/main/java/frc/team166/robot/Robot.java]: Probably fix the spacing so we don't actually change these lines.

File: src/main/java/frc/team166/robot/RobotMap.java [hunk 3]
```diff
@@ -78,10 +72,12 @@
         public final static int RED_LED = 4;
         public final static int GREEN_LED = 5;
         public final static int BLUE_LED = 6;
+        public final static int LIFT_A = 0;
+        public final static int LIFT_B = 1;
     }
 
     public static class PreferenceStrings {
-        //changes preferences to strings
+        //changes preferences to stringsq
```
Comment (bot190) [inline src/main/java/frc/team166/robot/RobotMap.java]: woops.. Typo here

File: src/main/java/frc/team166/robot/subsystems/Lift.java [hunk 4]
```diff
@@ -129,6 +133,11 @@ private void raiseLift() {
         liftDrive.set(-0.75);
     }
 
+    private void lowerLift() {
+        //AJ changed to positive to test.
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Lift.java]: This comment is out of date, should be removed.

File: src/main/java/frc/team166/robot/subsystems/Lift.java [hunk 5]
```diff
@@ -186,14 +195,15 @@ public Command RaiseLiftALittle() {
...
                 disengageBrake();
-                liftDrive.set(0.5);
+                liftDrive.set(0.9);
             }
 
             @Override
             protected void execute() {
-
+                liftDrive.set(0.5);
+                System.out.println("Running! " + liftDrive.get());
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Lift.java]: No bueno. Lets remove this println so we don't get timeouts

File: src/main/java/frc/team166/robot/subsystems/Lift.java [hunk 6]
```diff
@@ -266,6 +277,7 @@ protected void execute() {
                 } else {
                     liftDrive.set(elevatorControl);
                 }
+                System.out.println(liftDrive.get() + "Bottom Switch: " + bottomLimitSwitch.get());
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Lift.java]: Get rid of this.

File: src/main/java/frc/team166/robot/subsystems/Lift.java [hunk 7]
```diff
@@ -285,6 +297,30 @@ protected void interrupted() {
         };
     }
 
+    public Command UpSpecified(double inches) {
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Lift.java]: Rename to LiftByInches or something similar

File: src/main/java/frc/team166/robot/subsystems/Lift.java [hunk 8]
```diff
@@ -285,6 +297,30 @@ protected void interrupted() {
...
+        return new SubsystemCommand(this) {
+            double upSpecifiedIncrement = liftEncoder.get();
+            @Override
+            protected void initialize() {
+                disengageBrake();
+                upSpecifiedIncrement = liftEncoder.get();
+            }
+            @Override
+            protected void execute() {
+                if (upSpecifiedIncrement + 1 <=1) {
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Lift.java]: You can start the liftDrive in initialize, leave execute empty. Then add an end() which calls liftDrive.set(0)

File: src/main/java/frc/team166/robot/subsystems/Lift.java [hunk 9]
```diff
@@ -285,6 +297,30 @@ protected void interrupted() {
...
+                    liftDrive.set(.5);
+                } else liftDrive.set(0);
+            }
+            
+            @Override
+            protected boolean isFinished() {
+                if (upSpecifiedIncrement + 1 >=1) {
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Lift.java]: This needs to look at the current encoder distance with liftEncoder.get() and compare it to the initial height + inches

File: src/main/java/frc/team166/robot/subsystems/Lift.java [hunk 10]
```diff
@@ -324,6 +360,46 @@ protected boolean isFinished() {
...
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return !bottomLimitSwitch.get();
+            }
+        };
+    }
+
+    public Command LowerLiftForTime() {
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Lift.java]: This should probably accept a double timeout as an argument and use that for the setTimeout() call.

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java [hunk 11]
```diff
@@ -249,22 +255,17 @@ public Command CubeEject() {
...
             protected void initialize() {
-                setTimeout(Preferences.getInstance().getDouble(RobotMap.PreferenceStrings.CUBE_EJECT_WAIT_TIME, 2.0));
+                setTimeout(Preferences.getInstance().getDouble(RobotMap.PreferenceStrings.CUBE_EJECT_WAIT_TIME, 1));
                 String gameData;
                 gameData = DriverStation.getInstance().getGameSpecificMessage();
 
                 if (gameData.length() > 0) {
                     if (gameData.charAt(0) == 'R') {
-                        setMotorsToDischarge();
+                        setMotorsToIntake();
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Manipulator.java]: Huh?

File: src/main/java/frc/team166/robot/Robot.java [hunk 12]
```diff
@@ -7,8 +7,11 @@
 
 package frc.team166.robot;
 
+import javax.lang.model.util.ElementScanner6;
```
Comment (bot190) [inline src/main/java/frc/team166/robot/Robot.java]: It snuck in again..

**Review by bot190 — COMMENTED** [review]
Turns out there are quite a few changes..

File: src/main/java/frc/team166/robot/subsystems/Lift.java (hunk 5 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Lift.java]: done

File: src/main/java/frc/team166/robot/subsystems/Lift.java (hunk 4 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Lift.java]: done

File: src/main/java/frc/team166/robot/RobotMap.java (hunk 3 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/RobotMap.java]: done

File: src/main/java/frc/team166/robot/Robot.java (hunk 2 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/Robot.java]: done

File: src/main/java/frc/team166/robot/Robot.java (hunk 1 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/Robot.java]: done

File: src/main/java/frc/team166/robot/Robot.java (hunk 12 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/Robot.java]: done

File: src/main/java/frc/team166/robot/subsystems/Lift.java (hunk 6 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Lift.java]: done

File: src/main/java/frc/team166/robot/subsystems/Lift.java (hunk 7 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Lift.java]: done

File: src/main/java/frc/team166/robot/subsystems/Lift.java (hunk 8 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Lift.java]: done

File: src/main/java/frc/team166/robot/subsystems/Lift.java (hunk 9 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Lift.java]: done

File: src/main/java/frc/team166/robot/subsystems/Lift.java (hunk 10 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Lift.java]: done

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java (hunk 11 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Manipulator.java]: doneeeeee

File: src/main/java/frc/team166/robot/Robot.java [hunk 13]
```diff
@@ -7,8 +7,11 @@
 
 package frc.team166.robot;
 
+//import javax.lang.model.util.ElementScanner6;
```
Comment (bot190) [inline src/main/java/frc/team166/robot/Robot.java]: Delete this, don't just comment it..

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java [hunk 14]
```diff
@@ -362,23 +365,24 @@ public Command DeployManipulatorWithJoystick() {
...
 
-            @Override
-            protected boolean isFinished() {
-                return false;
-            }
-
             @Override
             protected void execute() {
                 rotation = Math.pow(Robot.m_oi.xBoxTempest.getY(Hand.kLeft), 2);
+                // rotation = Math.pow(Robot.m_oi.xBoxTempest.getY(Hand.kLeft), 2.5);
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Manipulator.java]: This should probably just be removed

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java [hunk 15]
```diff
@@ -336,7 +338,8 @@ protected void end() {
     }
 
     // public Command CubePickupWithLights(int blinkCount) {
-    //     return new CommandChain("Cube Pickup with Lights").then(CubePickup()).then(Robot.led.blinkGreen(blinkCount));
+    // return new CommandChain("Cube Pickup with
+    // Lights").then(CubePickup()).then(Robot.led.blinkGreen(blinkCount));
```
Comment (bot190) [inline src/main/java/frc/team166/robot/subsystems/Manipulator.java]: Just remove this command?

**Review by bot190 — APPROVED** [review]
Looking good, just some things that can be removed

File: src/main/java/frc/team166/robot/Robot.java (hunk 13 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/Robot.java]: done

File: src/main/java/frc/team166/robot/subsystems/Manipulator.java (hunk 14 above)
Comment (strimper13) [reply src/main/java/frc/team166/robot/subsystems/Manipulator.java]: done

**Pull request merged**